### PR TITLE
Non learnable featurizers

### PR DESCRIPTION
**Pull request by thodrek**
Changes:

1) Changed featurizer abstract class to support fixed-weight featurizers.
2) Changed repair.py and learn.py to use new featurizers.

### CODE REVIEW

File: repair/featurize/featurizer.py [hunk 1]
```diff
@@ -5,9 +5,11 @@
 class Featurizer:
     __metaclass__ = ABCMeta
 
-    def __init__(self, name):
```
Comment (minafarid) [inline repair/featurize/featurizer.py:8]: I am curious why we removed the `name` from the constructor and moved it to `specific_setup`?

File: repair/learn/learn.py [hunk 2]
```diff
@@ -19,31 +19,45 @@ class TiedLinear(torch.nn.Module):
...
+    def concat_weights(self):
+        self.W = torch.cat([t.expand(self.output_dim, -1) for t in self.weight_list],-1)
+        if self.bias_flag:
+            self.B = torch.cat([t.expand(self.output_dim, -1) for t in self.bias_list],-1)
 
     def forward(self, X, index, mask):
+        # Concats different featurizer weights - need to call during every pass
+        self.concat_weights()
         output = X.mul(self.W)
         if self.bias is not None:
```
Comment (minafarid) [inline repair/learn/learn.py]: This needs to change to `bias_flag` or check the `len(self.bias_list)`

File: repair/learn/learn.py [hunk 3]
```diff
@@ -154,6 +168,4 @@ def get_featurizer_weights(self, feat_info):
                 'weights': this_weight,
                 'size': f[1]
             }
-            begin = begin+f[1]
-        return report
-
+        return report
```
Comment (minafarid) [inline repair/learn/learn.py]: minor: please add a new line at the end of the file.

**Review by minafarid — CHANGES_REQUESTED** [review]
FYI, I ran this PR on the `hospital` example and this is the results I got

`INFO:root:Precision = 0.98, Recall = 0.50, Repairing Recall = 0.58, F1 = 0.66, Repairing F1 = 0.73, Detected Errors = 438, Total Errors = 509, Correct Repairs = 255, Total Repairs = 259, Total Repairs (Grdth present) = 259`

File: repair/featurize/featurizer.py (hunk 1 above)
Comment (thodrek) [reply repair/featurize/featurizer.py:8]: What we had before was overriding the constructor and hence it was not possible to call "learnable=True" etc from the constructor. It had to be explicitly defined for each featurizer. With this change we do not need to define the same constructor over and over again.

**Review by minafarid — APPROVED** [review]
LGTM

**Pull request merged**